Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -392,6 +402,23 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
     Cookie cookie;
     cookie.name = WTFMove(options.name);
     cookie.value = WTFMove(options.value);
+
+    if (cookie.name.isEmpty()) {
+        if (cookie.value.isEmpty()) {
+            promise->reject(Exception { ExceptionCode::TypeError, "The cookie name and value must not both be empty."_s });
+            return;
+        }
+        if (cookie.value.contains('=')) {
+            promise->reject(Exception { ExceptionCode::TypeError, "The cookie name and value must not both be set from the 'value' field."_s });
+            return;
+        }
+    }
+
+    if (containsInvalidCharacters(cookie.name) || containsInvalidCharacters(cookie.value)) {
```

Comment:
It would be nice to split these 2 checks so we can provide a better promise rejection message.

Reply:
Fixed!